shared/middleware/helpers/core.rb

```diff
@@ -46,9 +46,19 @@ def not_found()
 end
 
 def too_large()
-  halt(413, "Payload too large\n")
```

Comment:
hm, presumably the nearby methods work when called somewhere (that may be presuming too much though :)). And that somewhere is calling halt instead of raising exceptions for a reason... if you have a debugger you could track down the exception handler by putting a breakpoint in one of the methods in TooLarge (or if you don't have a debugger you could raise a RuntimeError in there).